### DISCUSSION TITLE
Determine if project has `required-dev` packages

### DISCUSSION THREAD

**Discussion by bilogic** [composer/composer#11457] [Q&A / Support]
Hi,

Is there a way for a project to determine if it has installed/updated itself with `--no-dev`?
Meaning, in my project:

1. For `composer update`, when I run `MyComposer::hasDevPackages()`, it returns `true`
2. For `composer update --no-dev`, `MyComposer::hasDevPackages()` returns `false`

**Comment by Seldaek** [comment]
I'd suggest using https://getcomposer.org/doc/07-runtime.md#knowing-whether-package-x-or-virtual-package-is-present to check if the package you need is present, but if what you need is really the dev state, you can use `InstalledVersions::getRootPackage()['dev']` which is a bool
